pulsar/producer_partition.go

```diff
@@ -361,19 +407,48 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 	msg := request.msg
 
 	payload := msg.Payload
+
 	var schemaPayload []byte
 	var err error
-	if p.options.Schema != nil {
-		schemaPayload, err = p.options.Schema.Encode(msg.Value)
+	if msg.Value != nil && msg.Payload != nil {
+		p.log.Error("Can not set Value and Payload both")
+		return
+	}
+
+	if p.options.DisableMultiSchema {
+		if msg.Schema == nil && p.options.Schema == nil && !msg.Schema.GetSchemaInfo().isSame((p.options.Schema).GetSchemaInfo()) {
```

Comment:
Can this be simplified or a comment added. I'm having a difficult time understanding.